dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/importer/ValidatingEventChecker.java

```diff
@@ -25,48 +25,49 @@
  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
  * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
-package org.hisp.dhis.dxf2.events.importer.delete.validation;
-
-import static com.google.common.base.Preconditions.checkNotNull;
-import static java.util.Collections.emptyList;
-import static org.hisp.dhis.dxf2.events.importer.ImportStrategyUtils.isDelete;
-import static org.hisp.dhis.importexport.ImportStrategy.DELETE;
+package org.hisp.dhis.dxf2.events.importer;
 
+import java.util.Collections;
 import java.util.List;
-import java.util.Map;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import lombok.RequiredArgsConstructor;
 
+import org.hisp.dhis.dxf2.common.ImportOptions;
 import org.hisp.dhis.dxf2.events.event.Event;
-import org.hisp.dhis.dxf2.events.importer.Checker;
-import org.hisp.dhis.dxf2.events.importer.EventChecking;
 import org.hisp.dhis.dxf2.events.importer.context.WorkContext;
 import org.hisp.dhis.dxf2.importsummary.ImportSummary;
 import org.hisp.dhis.importexport.ImportStrategy;
-import org.springframework.stereotype.Component;
 
-/**
- * @author maikel arabori
- */
-@Component( "eventsDeleteValidationFactory" )
-public class DeleteValidationFactory implements EventChecking
+@RequiredArgsConstructor
+public abstract class ValidatingEventChecker implements EventChecker
 {
-    private final Map<ImportStrategy, List<Class<? extends Checker>>> eventDeleteValidatorMap;
 
-    public DeleteValidationFactory( final Map<ImportStrategy, List<Class<? extends Checker>>> eventDeleteValidatorMap )
-    {
-        checkNotNull( eventDeleteValidatorMap );
-        this.eventDeleteValidatorMap = eventDeleteValidatorMap;
-    }
+    private final List<? extends Checker> checkers;
+
+    private final EventImporterValidationRunner validationRunner;
 
     @Override
     public List<ImportSummary> check( final WorkContext ctx, final List<Event> events )
     {
-        final ImportStrategy importStrategy = ctx.getImportOptions().getImportStrategy();
-
-        if ( isDelete( importStrategy ) )
-        {
-            return new ValidationRunner( ctx, events ).run( eventDeleteValidatorMap.get( DELETE ) );
-        }
+        return Optional.ofNullable( ctx )
+            .map( WorkContext::getImportOptions )
+            .map( ImportOptions::getImportStrategy )
+            .filter( this::isSupported )
+            .map( importStrategy -> EventImporterValidationRunner.ValidationRunnerPayload.of(
+                ctx,
+                events,
+                checkers ) )
+            .map( validationRunner::run )
+            .orElse( Collections.emptyList() );
```

Comment:
Streaming one way into and out of the actual business logic is taking streams a bit too far for my taste as it make it hard to see what the essential thing is we want to do.

Reply:
No streams here, just Optional, however I got your point, I'll change to something more readable